api/arceos_posix_api/src/imp/fs.rs

```diff
@@ -196,13 +196,51 @@ pub unsafe extern "C" fn ax_getcwd(buf: *mut c_char, size: usize) -> *mut c_char
 /// If new exists, it is first removed.
 ///
 /// Return 0 if the operation succeeds, otherwise return -1.
-#[no_mangle]
-pub unsafe extern "C" fn ax_rename(old: *const c_char, new: *const c_char) -> c_int {
-    ax_call_body!(ax_rename, {
+pub fn sys_rename(old: *const c_char, new: *const c_char) -> c_int {
+    syscall_body!(sys_rename, {
         let old_path = char_ptr_to_str(old)?;
         let new_path = char_ptr_to_str(new)?;
-        debug!("ax_rename <= old: {:?}, new: {:?}", old_path, new_path);
-        axstd::fs::rename(old_path, new_path)?;
+        debug!("sys_rename <= old: {:?}, new: {:?}", old_path, new_path);
+        axfs::api::rename(old_path, new_path)?;
         Ok(0)
     })
 }
+
+/// Get file metadata by `fd` and write into `buf`.
+///
+/// Return 0 if success.
+pub unsafe fn sys_fstat(fd: c_int, buf: *mut ctypes::stat) -> c_int {
+    debug!("sys_fstat <= {} {:#x}", fd, buf as usize);
+    syscall_body!(sys_fstat, {
+        if buf.is_null() {
+            return Err(LinuxError::EFAULT);
+        }
+        #[cfg(feature = "fd")]
```

Comment:
Since the feature `fs` depends on the feature `fd`, the `#[cfg()]` block can be removed.

Reply:
So this means: if a user app want to do something like: `fstat(0/1/2, buf)`, the user app must have feature `fs`, since `fstat` depends on `fs` feature?

Reply:
It's rare to call `fstat` on stdin/stdout/stderr?